include/alpaka/math/Complex.hpp

```diff
@@ -579,4 +581,366 @@ namespace alpaka
     } // namespace internal
 
     using internal::Complex;
+
+#if defined(ALPAKA_ACC_SYCL_ENABLED) || defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+
+    namespace math::trait
+    {
+
+        //! The abs trait specialization for complex types.
+        template<typename TAcc, typename T>
+        struct Abs<TAcc, Complex<T>>
+        {
+            //! Take context as original (accelerator) type, since we call other math functions
+            template<typename TCtx>
+            ALPAKA_FN_ACC auto operator()(TCtx const& ctx, Complex<T> const& arg)
```

Comment:
`ALPAKA_FN_ACC` must be `ALPAKA_FN_HOST_ACC` to match the old `__device__ __host__` in cases where code is compiler with the cmake flag `alpaka_ACC_GPU_CUDA_ONLY` the device compiler nvcc can not deduce the return type.

@chillenzer will fix ist^^